test/server/proxy-test.js

```diff
@@ -2606,5 +2664,43 @@ describe('Proxy', () => {
                 testRedirectRequestStatusCode(202, false)
             ]);
         });
+
+        it('Should ensure a trailing slash on location header (GH-1426)', () => {
+            function getExpectedProxyUrl (proxiedCase) {
+                const proxiedUrl = urlUtils.getProxyUrl(proxiedCase.url, {
+                    proxyHostname: '127.0.0.1',
+                    proxyPort:     1836,
+                    sessionId:     session.id
+                });
+
+                return proxiedUrl + (proxiedCase.shoudAddTrailingSlash ? '/' : '');
+            }
+
+            const proxyUrls = ENSURE_URL_TRAILING_SLASH_TEST_CASES.map(testCase => {
+                return {
+                    proxiedUrl:            proxy.openSession(testCase.url, session),
+                    shoudAddTrailingSlash: testCase.shoudAddTrailingSlash,
+                    url:                   testCase.url
+                };
+            });
+
+            const testPageRequest = proxiedCase => {
+                const encodedUrl = encodeURIComponent(proxiedCase.proxiedUrl);
+                const options    = {
+                    url: 'http://127.0.0.1:2000/redirect/' + encodedUrl,
+
+                    resolveWithFullResponse: true,
+                    followRedirect:          false,
+                    simple:                  false,
+                };
+
+                return request(options)
+                    .then(res => {
+                        expect(res.headers['location']).eql(getExpectedProxyUrl(proxiedCase));
+                    });
+            };
+
+            return Promise.all(proxyUrls.map(testPageRequest));
```

Comment:
Rename `testPageRequest` to `testLocationHeaderValue`.

Combine `proxyUrls` with `testPageRequest`.

Returned Promise should looks like `return Promise.all(ENSURE_URL_TRAILING_SLASH_TEST_CASES.map(testLocationHeaderValue))`